### PR TITLE
Feature/us676841 environment metadata

### PR DESCRIPTION
**Pull request by uppoju**
## Description
Generating metadata for environment bundle
## Links
Link Type   | Link
------      | ------
Rally Issue | 
Func Spec   | 
## Checklist
- [x] Pull Request Created
- [ ] Code Review Completed
- [ ] Veracode scan results addressed
- [ ] SonarQube scan results addressed
- [ ] TPSR has been submitted (if applicable) 
- [ ] Unit tests have been added
- [ ] Integration/Functional test cases have been written and automated
- [ ] Func Spec has been updated as needed
- [ ] Rally Issue(s) are in an engineering completed state

### CODE REVIEW

File: config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleMetadataBuilder.java [hunk 1]
```diff
@@ -75,6 +75,18 @@ public BundleMetadata build(final AnnotatedBundle annotatedBundle, final Bundle
         }
     }
 
+    public BundleMetadata buildEnvironmentMetadata(final List<Entity> entities, ProjectInfo projectInfo){
```
[the remark this answers is not in the document]
Comment (uppoju) [reply config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleMetadataBuilder.java:78]: not sure if this is valid case as entities may be different for different environments

[the remark this answers is not in the document]
Comment (uppoju) [reply config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleMetadataBuilder.java:78]: And environment bundle does not include any entities from dependency bundles

File: environment-creator-application/src/main/java/com/ca/apim/gateway/cagatewayconfig/EnvironmentCreatorApplication.java [hunk 2]
```diff
@@ -88,7 +88,9 @@ void run() {
                 templatizedBundleFolderPath,
                 environmentConfigurationFolderPath,
                 APPLICATION,
-                "_0_env.req.bundle"
+                "_0_env.req.install.bundle",
```
Comment (nikhilahuja) [inline environment-creator-application/src/main/java/com/ca/apim/gateway/cagatewayconfig/EnvironmentCreatorApplication.java:91]: Is `gw7` build running successfully? Also, all JUnits are working after this change?

File: config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleEntityBuilder.java [hunk 3]
```diff
@@ -52,13 +52,18 @@
 
     public Map<String, BundleArtifacts> build(Bundle bundle, BundleType bundleType,
                                               Document document, ProjectInfo projectInfo) {
+        return build(bundle, bundleType, document, projectInfo, false);
+    }
+
+    public Map<String, BundleArtifacts> build(Bundle bundle, BundleType bundleType,
+                                              Document document, ProjectInfo projectInfo, boolean generateMetadata) {
```
Comment (nikhilahuja) [inline config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleEntityBuilder.java:59]: Why is `generateMetadata` is required? Is there a case when we don't want to generate metadata but create Env bundle?

File: environment-creator-application/src/main/java/com/ca/apim/gateway/cagatewayconfig/EnvironmentCreatorApplication.java (hunk 2 above)
Comment (uppoju) [reply environment-creator-application/src/main/java/com/ca/apim/gateway/cagatewayconfig/EnvironmentCreatorApplication.java:91]: yes

File: config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleEntityBuilder.java (hunk 3 above)
Comment (uppoju) [reply config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleEntityBuilder.java:59]: this same method is called by Full bundle task for creating environment entities

**Pull request merged**